packages/block-editor/src/components/block-list/use-block-props/use-zoom-out-mode-exit.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useRefEffect } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../../store';
+import { unlock } from '../../../lock-unlock';
+
+/**
+ * Allows Zoom Out mode to be exited by double clicking in the selected block.
+ *
+ * @param {string} clientId Block client ID.
+ */
+export function useZoomOutModeExit( clientId ) {
+	const { isZoomOutMode, isBlockSelected } = unlock(
+		useSelect( blockEditorStore )
+	);
+	const { __unstableSetEditorMode, selectBlock } = unlock(
+		useDispatch( blockEditorStore )
+	);
+
+	return useRefEffect(
+		( node ) => {
+			function onDoubleClick( event ) {
+				// Don't select a block if it's already handled by a child
+				// block.
+				if ( isZoomOutMode() && ! event.defaultPrevented ) {
+					// Prevent focus from moving to the block.
+					event.preventDefault();
+
+					__unstableSetEditorMode( 'edit' );
+					selectBlock( clientId );
```

Comment:
This should already be selected due to [the focus handler in block props selecting the block](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/block-list/use-block-props/use-focus-handler.js#L57). Could we remove the `selectBlock` and leave a comment about where the block selection happens?